src/server/lib/okta/api/users.ts

```diff
@@ -174,44 +174,42 @@ export const reactivateUser = async (
 };
 
 /**
- * @name generateResetPasswordToken
- * @description Generates a reset password token for a user
+ * @name resetPassword
  *
- * https://developer.okta.com/docs/reference/api/users/#reset-password
+ * YOU PROBABLY DON'T WANT TO USE THIS TO RESET PASSWORD
  *
- * Generates a one-time token (OTT) that can be used to reset a user's password.
- * The OTT link can be automatically emailed to the user or returned to the API
- * caller and distributed using a custom flow.
+ * USE `resetPassword` method from the authentication API (./authentication.ts)
+ *  OR `forgotPassword` method method below
  *
- * This operation will transition the user to the status of RECOVERY and the
- * user will not be able to login or initiate a forgot password flow until they
- * complete the reset flow.
+ * AS THEY DO NOT CHANGE THE USER'S STATE
  *
- * Returns an empty object by default. If sendEmail is false, returns a link
- * for the user to reset their password.
+ * USE THIS ONLY IF YOU KNOW WHY YOU NEED TO
  *
- * @param id accepts the Okta user ID, email address (login) or login shortname (as long as it is unambiguous)
- * @param sendEmail Sends an activation email to the user if true
+ * Currently used by the reset password flow to set a placeholder password for SOCIAL
+ * users who don't have a password. This method is the only way to change their provider
+ * from SOCIAL to OKTA so that a password can be set and then reset.
  *
- * @returns Promise<TokenResponse | void>
+ * Generates a one-time token (OTT) that can be used to reset a user's password.
+ *
+ * This operation will transition the user to the status of RECOVERY and the user
+ * will not be able to login or initiate a forgot password flow until they complete
+ * the reset flow.
+ *
+ * https://developer.okta.com/docs/reference/api/users/#reset-password
+ *
+ * @param id Okta user Id
+ * @returns Promise<string>
  */
-export const generateResetPasswordToken = async (
-  id: string,
-  sendEmail = false,
-): Promise<TokenResponse | void> => {
+export const resetPassword = async (id: string): Promise<string> => {
```

Comment:
could the method name reflect the fact that this is generally a dangerous method to use if you haven't read the included comment? something along the lines of: `resetPasswordAndDangerouslySetState` or something?

Reply:
changed the name to `dangerouslyResetPassword`